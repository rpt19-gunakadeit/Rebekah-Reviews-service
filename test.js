import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';


////////////// TEST THE APP COMPONENT ////////////////////
import Reviews from './client/components/reviews.jsx';



test('<Reviews /> renders correctly', () => {
  const component = renderer.create(<Reviews />).toJSON();
  expect(component).toMatchSnapshot();
})



test('<Reviews /> initally renders the inner <Stars /> component, but not the <SummaryReviews /> component', () => {
  const wrapper = mount(<Reviews />);
  expect(wrapper.find(Stars).length).toEqual(1);
  expect(wrapper.find(SummaryReviews).length).toEqual(0);
  expect(wrapper.find('i.up').length).toEqual(0);
  expect(wrapper.find('i.down').length).toEqual(1);
  wrapper.unmount();
})

test('<Reviews /> default state to show short reviews is false', () => {
  const component = shallow(<Reviews/>);
  expect(component.state('threeReviews')).toBe(false);
})

test('<Reviews /> simulates click events to show short reviews', () => {
  const component = shallow(<Reviews />);
  expect(component.state('threeReviews')).toEqual(false);
  expect(component.find('SummaryReviews').length).toEqual(0);

  component.find('div#reviews-header').simulate('click');
  expect(component.state('threeReviews')).toEqual(true);
  expect(component.find('SummaryReviews').length).toEqual(1);
});

test('<Reviews /> correctly renders elements based on its state', () => {
  const component = shallow(<Reviews />);
  expect(component.find('span').at(0).text()).toEqual('Reviews (0)');

  component.setState({ reviews: [{review: 1}, {review: 2}] })
  expect(component.find('span').at(0).text()).toEqual('Reviews (2)');
});

test('<Reviews /> passes props to the <Stars /> component', () => {
  let component = mount(<Reviews />);
  expect(component.find(Stars).length).toEqual(1);
  let starsComponent = component.find(Stars);
  expect(starsComponent.prop('numStars')).toEqual(null);

  component.setState({ avgStars: 4 });
  expect(component.find(Stars).prop('numStars')).toEqual(4);
  component.unmount();
})




////////////// TEST THE STARS COMPONENT ////////////////////
import Stars from './client/components/stars.jsx';
test('<Stars /> renders correctly', () => {
  const tree = renderer.create(<Stars />).toJSON();
  expect(tree).toMatchSnapshot();
})



////////////// TEST THE FIT RANGE COMPONENT ////////////////////
import FittingRange from './client/components/fittingRange.jsx';
test('<FittingRange /> renders correctly', () => {
  const tree = renderer.create(<FittingRange />).toJSON();
  expect(tree).toMatchSnapshot();
})



////////////// TEST THE SUMMARYREVIEWS COMPONENT ////////////////////
import SummaryReviews from './client/components/summaryReviews.jsx';
test('<SummaryReviews /> renders correctly', () => {
  const tree = renderer.create(<SummaryReviews />).toJSON();
  expect(tree).toMatchSnapshot();
})

test('<SummaryReviews /> renders the inner <Stars /> component', () => {
  const wrapper = mount(<SummaryReviews/>);
  expect(wrapper.find(Stars).length).toEqual(1);
})


////////////// TEST THE SHORTREVIEW COMPONENT ////////////////////
import ShortReview from './client/components/shortReview.jsx';
const shortData = {
  title: 'pariatur culpa aute quis incididunt',
  body:
   'Eu ut occaecat non fugiat. Labore non fugiat laboris amet pariatur voluptate. Ex voluptate velit cupidatat veniam occaecat. Sint enim incididunt eu aute adipisicing est ipsum minim.\nIpsum elit est in deserunt tempor Lorem veniam commodo.',
  stars: 5,
  date: '2020-02-06T01:20:00.000Z',
  location: 'Lake Jonathanmouth, Nebraska',
  user: 'Raoul54498'
};

test('<ShortReview /> renders correctly', () => {
  const tree = renderer.create(<ShortReview review={shortData}/>).toJSON();
  expect(tree).toMatchSnapshot();
})

test('<ShortReview /> renders the inner <Stars /> component', () => {
  const wrapper = mount(<ShortReview review={shortData}/>);
  expect(wrapper.find(Stars).length).toEqual(1);
})



////////////// TEST THE FULLREVIEWS COMPONENT ////////////////////
import FullReviews from './client/components/fullReviews.jsx';
const productDetails = {
  name: 'Nike Blazers',
  price: 185,
  salePrice: 150,
  thumbnail:'https://static.nike.com/a/images/t_default/eric5lwitzffpoisq0rj/blazer-mid-77-vintage-shoe'
}
test('<FullReviews /> renders correctly', () => {
  const tree = renderer.create(<FullReviews product={productDetails}/>).toJSON();
  expect(tree).toMatchSnapshot();
})

test('<FullReviews /> renders the inner <Stars /> component', () => {
  const wrapper = mount(<FullReviews product={productDetails}/>);
  expect(wrapper.find(Stars).length).toEqual(1);
})



////////////// TEST THE LONGREVIEW COMPONENT ////////////////////
import LongReview from './client/components/longReview.jsx';
const longReviewData = {
  id: 598,
  product_id: 20,
  title: 'pariatur culpa aute quis incididunt',
  body:
   'Eu ut occaecat non fugiat. Labore non fugiat laboris amet pariatur voluptate. Ex voluptate velit cupidatat veniam occaecat. Sint enim incididunt eu aute adipisicing est ipsum minim.\nIpsum elit est in deserunt tempor Lorem veniam commodo.',
  stars: 5,
  date: '2020-02-06T01:20:00.000Z',
  size_rating: 1,
  comfort_rating: 1,
  durability_rating: null,
  location: 'Lake Jonathanmouth, Nebraska',
  userId: 915,
  user: 'Raoul54498'
}
test('<LongReview /> renders correctly', () => {
  const tree = renderer.create(<LongReview review={longReviewData}/>).toJSON();
  expect(tree).toMatchSnapshot();
})

test('<LongReview /> renders the inner <Stars /> & <FittingRange /> components', () => {
  const wrapper = mount(<LongReview review={longReviewData}/>);
  expect(wrapper.find(Stars).length).toEqual(1);
  expect(wrapper.find(FittingRange).length).toEqual(1);
})

