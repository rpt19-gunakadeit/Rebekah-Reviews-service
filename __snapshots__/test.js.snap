// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FittingRange /> renders correctly 1`] = `
<div
  className="fit-scales-column"
>
  <div
    className="slidecontainer"
  />
</div>
`;

exports[`<FullReviews /> renders correctly 1`] = `
<div
  className="reviewModal"
>
  <header
    id="modal-header"
  >
    <span
      className="exit small-product-info"
      onClick={[Function]}
    >
      <img
        className="thumbnail"
        src="https://static.nike.com/a/images/t_default/eric5lwitzffpoisq0rj/blazer-mid-77-vintage-shoe"
      />
      <div
        className="small-product-details"
      >
        <span
          className="detail"
        >
          Nike Blazers
        </span>
        <span
          className="detail"
        >
          $
          150
        </span>
        <span
          className="detail price"
        >
          $
          185
        </span>
      </div>
    </span>
    <span
      className="exit"
      onClick={[Function]}
    >
      ×
    </span>
  </header>
  <div
    id="modal-content"
  >
    <div
      className="full-reviews-summary"
    >
      <div>
        <span>
          <span
            className="star"
          >
            ★
          </span>
          <span
            className="star"
          >
            ★
          </span>
          <span
            className="star"
          >
            ★
          </span>
          <span
            className="star"
          >
            ★
          </span>
          <span
            className="star"
          >
            ★
          </span>
        </span>
      </div>
      <div
        className="num-reviews"
      >
        0
         REVIEWS
      </div>
    </div>
    <div
      className="fit-scales breaker-line"
    >
      <div
        className="slidecontainer"
      />
    </div>
    <div
      className="breaker-line"
    >
      <select
        defaultValue="date"
        onChange={[Function]}
      >
        <option
          value="upvotes"
        >
          Sort By: Most Helpful
        </option>
        <option
          value="date"
        >
          Sort By: Newest
        </option>
        <option
          value="stars"
        >
          Sort By: Highest to Lowest
        </option>
        <option
          value="starsLow"
        >
          Sort By: Lowest to Highest
        </option>
      </select>
    </div>
    <div
      className="full-revew-load-more"
    />
  </div>
</div>
`;

exports[`<LongReview /> renders correctly 1`] = `
<div
  className="breaker-line"
  id="long-review"
>
  <div
    className="long-review-stars"
  >
    <span>
      <span
        className="star-checked"
      >
        ★
      </span>
      <span
        className="star-checked"
      >
        ★
      </span>
      <span
        className="star-checked"
      >
        ★
      </span>
      <span
        className="star-checked"
      >
        ★
      </span>
      <span
        className="star-checked"
      >
        ★
      </span>
    </span>
    <div
      className="fit-scales-column"
    >
      <div
        className="slidecontainer"
      >
        <span
          className="fit-title"
        >
          Size
        </span>
        <input
          className="slider"
          max="3"
          min="1"
          readOnly={true}
          step=".1"
          type="range"
          value={1}
        />
        <div
          className="sliderLabels"
        >
          <span>
            Runs Small
          </span>
          <span>
            Runs Big
          </span>
        </div>
      </div>
      <div
        className="slidecontainer"
      >
        <span
          className="fit-title"
        >
          Comfort
        </span>
        <input
          className="slider"
          max="3"
          min="1"
          readOnly={true}
          step=".1"
          type="range"
          value={1}
        />
        <div
          className="sliderLabels"
        >
          <span>
            Uncomfortable
          </span>
          <span>
            Very Comfortable
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="long-review-details"
  >
    <span
      className="full-review-title"
    >
      <b>
        pariatur culpa aute quis incididunt
      </b>
    </span>
    <span
      className="full-review-body"
    >
      Eu ut occaecat non fugiat. Labore non fugiat laboris amet pariatur voluptate. Ex voluptate velit cupidatat veniam occaecat. Sint enim incididunt eu aute adipisicing est ipsum minim.
Ipsum elit est in deserunt tempor Lorem veniam commodo.
       
       [This review was collected as part of a promotion.]
    </span>
    <span
      className="full-review-sml-details"
    >
      Feb 05, 2020
        -  
      Raoul54498
        -  
      Lake Jonathanmouth, Nebraska
    </span>
    <span
      className="full-review-sml-details"
    >
      Use for: Everyday wear
    </span>
    <span
      id="review-rating"
    >
      <span
        className="arrow-icon arrow-up"
        onClick={[Function]}
      >
        ⇧
      </span>
      <span
        className="review-rating-num"
      />
      <span
        className="arrow-icon arrow-down"
        onClick={[Function]}
      >
        ⇩
      </span>
      <span
        className="review-rating-num"
      />
      <span
        className="flag-icon"
        onClick={[Function]}
      >
        ⚐
      </span>
    </span>
  </div>
</div>
`;

exports[`<Reviews /> renders correctly 1`] = `
<div
  className="breaker-line"
>
  <div
    id="reviews-header"
    onClick={[Function]}
  >
    <span>
      Reviews (
      0
      )
    </span>
    <div>
      <span
        className="stars"
      >
        <span>
          <span
            className="star"
          >
            ★
          </span>
          <span
            className="star"
          >
            ★
          </span>
          <span
            className="star"
          >
            ★
          </span>
          <span
            className="star"
          >
            ★
          </span>
          <span
            className="star"
          >
            ★
          </span>
        </span>
      </span>
      <span>
        <i
          className="arrow down"
        />
      </span>
    </div>
  </div>
</div>
`;

exports[`<ShortReview /> renders correctly 1`] = `
<div
  id="review-preview"
>
  pariatur culpa aute quis incididunt
  <div
    className="review-info"
  >
    <span
      className="stars"
    >
       
      <span>
        <span
          className="star-checked"
        >
          ★
        </span>
        <span
          className="star-checked"
        >
          ★
        </span>
        <span
          className="star-checked"
        >
          ★
        </span>
        <span
          className="star-checked"
        >
          ★
        </span>
        <span
          className="star-checked"
        >
          ★
        </span>
      </span>
       
    </span>
    <span
      className="review-user-date"
    >
      Raoul54498
       - 
      Feb 05, 2020
    </span>
  </div>
  <span>
    Eu ut occaecat non fugiat. Labore non fugiat laboris amet pariatur voluptate. Ex voluptate velit cup
    ...
  </span>
  <div>
    <span
      className="more-less"
      onClick={[Function]}
    >
      More
    </span>
  </div>
</div>
`;

exports[`<Stars /> renders correctly 1`] = `
<span>
  <span
    className="star"
  >
    ★
  </span>
  <span
    className="star"
  >
    ★
  </span>
  <span
    className="star"
  >
    ★
  </span>
  <span
    className="star"
  >
    ★
  </span>
  <span
    className="star"
  >
    ★
  </span>
</span>
`;

exports[`<SummaryReviews /> renders correctly 1`] = `
<div
  id="short-reviews"
>
  <div>
    <div
      id="short-reviews-header"
    >
      <span
        className="stars"
      >
         
        <span>
          <span
            className="star"
          >
            ★
          </span>
          <span
            className="star"
          >
            ★
          </span>
          <span
            className="star"
          >
            ★
          </span>
          <span
            className="star"
          >
            ★
          </span>
          <span
            className="star"
          >
            ★
          </span>
        </span>
         
      </span>
      0
       Stars
    </div>
    <div>
      <span
        className="write-review-link"
      >
        Write a Review
      </span>
    </div>
  </div>
  <div />
  <div
    className="more-reviews"
  >
    <span
      className="more-reviews-link"
      onClick={[Function]}
    >
      More Reviews
    </span>
  </div>
</div>
`;
